RealmSwift/Sync.swift

```diff
@@ -530,3 +553,490 @@ extension Realm {
         return SyncSession(for: rlmRealm)
     }
 }
+
+#if canImport(Combine)
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, macCatalyst 13.0, macCatalystApplicationExtension 13.0, *)
+public extension App {
+    /// :nodoc:
+    enum UserError: Error {
+        /// UserError.uncertainState - Fallback for optional errors  in Future calls. Should never happened while it's still a technically possible case.
+        case uncertainState
```

Comment:
`uncertainState` doesn't really tell the user anything about what's occurred, nor does it make sense for it to be a `UserError` since it is being used all over.

Ideally, we would use something like `RLMAppError.unknown`, preferably with an error message stating that the Promise as failed for an unknown reason.